date_test.go

```diff
@@ -308,3 +308,91 @@ func TestDateSQL(t *testing.T) {
 		t.Error("value was wrong")
 	}
 }
+
+func TestAddMonthsNoOverflow(t *testing.T) {
```

Comment:
no tests for multiple months being added or subtracted. also worth adding a test that adding 0 months *never* changes the date, maybe with a loop.

Reply:
I reworked the test to include all the months, added a test with 0 months, and added a test with random months added.